### PR TITLE
`cider-connect`: also look for Babashka nrepl servers

### PR DESCRIPTION
**Pull request by vemv**
bb nrepls look like this in `ps`: `bb --nrepl-server`

A couple other refinements included.

No changelog is needed given there are already improvements under the same category.

Cheers - V

### CODE REVIEW

File: cider.el [hunk 1]
```diff
@@ -1914,29 +1914,34 @@ Use `cider-ps-running-lein-nrepls-command' and
...
+                                                                           (replace-regexp-in-string "^n" "" s))))
+                                                              (seq-filter #'identity)
+                                                              car))
+                                      (port (thread-last (split-string (shell-command-to-string (concat "lsof -n -P -Fn -i -a -p " pid))
+                                                                       "\n")
+                                                         (seq-map (lambda (s)
+                                                                    (when (string-prefix-p "n" s)
+                                                                      (replace-regexp-in-string ".*:" "" s))))
+                                                         (seq-filter #'identity)
+                                                         (seq-filter (lambda (s)
```
Comment (vemv) [inline cider.el:1938]: Added some extra filtering, just in case.

File: cider.el [hunk 2]
```diff
@@ -1914,29 +1914,34 @@ Use `cider-ps-running-lein-nrepls-command' and
...
+        (thread-last non-lein-nrepl-pids
+                     (mapcar (lambda (pid)
+                               (let* ((directory (thread-last (split-string (shell-command-to-string (concat "lsof -a -d cwd -n -Fn -p " pid))
+                                                                            "\n")
+                                                              (seq-map (lambda (s)
+                                                                         (when (string-prefix-p "n" s)
+                                                                           (replace-regexp-in-string "^n" "" s))))
+                                                              (seq-filter #'identity)
+                                                              car))
+                                      (port (thread-last (split-string (shell-command-to-string (concat "lsof -n -P -Fn -i -a -p " pid))
```
[the remark this answers is not in the document]
Comment (bbatsov) [reply cider.el:1932]: Perhaps you can mention this in a comment for future reference.

File: cider.el [hunk 3]
```diff
@@ -1914,29 +1914,34 @@ Use `cider-ps-running-lein-nrepls-command' and
   (unless (eq system-type 'windows-nt)
     (let ((non-lein-nrepl-pids
            (thread-last (split-string
-                         (shell-command-to-string "ps u | grep java | grep -v leiningen | grep nrepl.cmdline")
+                         (shell-command-to-string "ps u | grep -E 'java|nrepl.cmdline|--nrepl-server' | grep -v -E 'leiningen|grep'")
```
Comment (bbatsov) [inline cider.el:1917]: Might be nice to mention in a comment example of things this command is supposed to match. (e.g.  using `clj` and `bb`)

**Comment by vemv** [comment]
Thanks! All done.

**Pull request merged**